lib/ddtrace/contrib/active_model_serializers/patcher.rb

```diff
@@ -0,0 +1,99 @@
+require 'ddtrace/ext/app_types'
+require 'ddtrace/ext/http'
+require 'ddtrace/contrib/active_support/notifications/subscriber'
+
+module Datadog
+  module Contrib
+    module ActiveModelSerializers
+      # Provides instrumentation for ActiveModelSerializers through ActiveSupport instrumentation signals
+      module Patcher
+        include Base
+        include ActiveSupport::Notifications::Subscriber
+
+        VERSION_REQUIRED = Gem::Version.new('0.9.0')
+        RENDER_MESSAGE = 'active_model_serializers.render'.freeze
+        EVENT_NAME_10 = 'render.active_model_serializers'.freeze
+        EVENT_NAME_09 = '!serialize.active_model_serializers'.freeze
+        register_as :active_model_serializers
+        option :tracer, default: Datadog.tracer
+        option :service_name, default: 'active_model_serializers'
+
+        on_subscribe do
+          subscribe(event_name, RENDER_MESSAGE, {}, configuration[:tracer], &method(:process))
+        end
+
+        class << self
+          def patch
+            return patched? if patched? || !compatible?
+
+            @ams10 = nil
+            @event_name = nil
+
+            subscribe!
+            configuration[:tracer].set_service_info(
+              configuration[:service_name],
+              'active_model_serializers',
+              Ext::AppTypes::WEB
+            )
+
+            @patched = true
+          end
+
+          def patched?
+            return @patched if defined?(@patched)
+            @patched = false
+          end
+
+          def process(span, _, _, payload)
+            span.service = configuration[:service_name]
+
+            res = resource(payload[:serializer])
+            span.resource = res
+            span.set_tag('active_model_serializers.serializer', res)
+
+            span.span_type = Datadog::Ext::HTTP::TEMPLATE
+
+            # Will be nil in 0.9
+            span.set_tag('active_model_serializers.adapter', payload[:adapter].class) unless payload[:adapter].nil?
+          end
+
+          def event_name
+            return @event_name unless @event_name.nil?
+
+            @event_name = if ams10?
+                            EVENT_NAME_10
+                          else
+                            EVENT_NAME_09
+                          end
+          end
+
+          private
+
+          def resource(serializer)
+            if ams10?
+              serializer.name
+            else
+              serializer
+            end
+          end
+
+          def ams10?
```

Comment:
Name of this function is a little misleading; thought it implied a specific major version number. Could we change this name to something more that reflects "AMS 0.10 or greater"?

Alternatively, maybe it'd be more clear to communicate intent with these function names e.g. `use_render_event_name?` or `use_serializer_name?` that are based on a common `ams10_or_newer?`.

The latter suggestion is just food for thought; if you think the prior is better, that's probably fine too.